gadgets/profile_blockio/program.bpf.c

```diff
@@ -0,0 +1,170 @@
+// SPDX-License-Identifier: GPL-2.0
+// Copyright (c) 2020 Wenbo Zhang
+#include <vmlinux.h>
+#include <bpf/bpf_helpers.h>
+#include <bpf/bpf_core_read.h>
+#include <bpf/bpf_tracing.h>
+#include <gadget/bits.bpf.h>
+#include <gadget/core_fixes.bpf.h>
+#include <gadget/macros.h>
+
+#define MAX_ENTRIES 10240
+#define DISK_NAME_LEN 32
+#define MAX_SLOTS 27
+
+#define MINORBITS 20
+#define MINORMASK ((1U << MINORBITS) - 1)
+
+#define MKDEV(ma, mi) (((ma) << MINORBITS) | (mi))
+
+const volatile bool filter_cg = false;
+const volatile bool targ_per_disk = false;
+const volatile bool targ_per_flag = false;
+const volatile bool targ_queued = false;
+const volatile bool targ_ms = false;
+const volatile bool filter_dev = false;
+const volatile __u32 targ_dev = 0;
+
+extern int LINUX_KERNEL_VERSION __kconfig;
+
+struct hist_key {
+	__u32 cmd_flags;
+	__u32 dev;
+};
+
+struct hist_value {
+	__u32 slots[MAX_SLOTS];
+};
```

Comment:
Can it be a structure provided by Inspektor Gadget? In the end, all histograms share the same value structure. (MAX_SLOTS could be configurable). 

If we define this structure ourselves, we'll have a better control when unmarshaling it.